### ISSUE TITLE
Can't use vpn and karma-firewall at same time

### ISSUE THREAD

**Issue by Justauser0** [StarGW-net/karma-firewall#6]

**Comment by StarGW-net** [comment]
karma-firewall uses a VPN to firewall traffic. Android only allows one VPN at a time. So unfortunately its not possible.

**Comment by Justauser0** [comment]
> karma-firewall uses a VPN to firewall traffic. Android only allows one VPN at a time. So unfortunately its not possible.

Instead of using vpn could we use proxy firewall

**Comment by Justauser0** [comment]
> > karma-firewall uses a VPN to firewall traffic. Android only allows one VPN at a time. So unfortunately its not possible.
> 
> Instead of using vpn firewall could we use proxy firewall

Is it possible ?

**Comment by StarGW-net** [comment]
No, I do not think a proxy firewall would be possible. Not sure how it would work, and it would probably need to see all the traffic which is not something we want.

**Comment by Gabr-F** [comment]
Not being to use a vpn unfortunately is a big limitation. The fault of Google for the most part, true, but if there's a way to workaround the problem it would be better to do it.

Yes, it probably would require monitoring the traffic, but since now the app is on F-Droid that's much less an issue (and you could do it only when it's required for an enabled feature).

See (https://github.com/M66B/NetGuard/blob/master/FAQ.md#user-content-faq3)[here] and (https://github.com/M66B/NetGuard/blob/master/FAQ.md#user-content-faq54)[here] how Netguard does it (it simply supports proxies).

**Comment by VA1DER** [comment]
The best way I can think of would be to combine Karma and wireguard into one app.  This would be a lot of work, but you'd end up with an innovative solution, and one that would get a lot of attention.

**Comment by JoaoVictorVP** [comment]
@Gabr-F 
Sure?

> If the VPN application is using the [VPN service](http://developer.android.com/reference/android/net/VpnService.html), then no, because NetGuard needs to use this service. Android allows only one application at a time to use this service.

I think they have a workaround more than a literal support for other apps with VPN services (since they indeed are using it).
